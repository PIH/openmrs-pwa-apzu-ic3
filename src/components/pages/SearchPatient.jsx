import React from 'react';
import { PatientSearch } from '@openmrs/react-components';

class SearchPatient extends React.Component {

    constructor(props) {
        super(props);
        this.columnDefs =  [
            {headerName: 'patientUuid', hide: true, valueGetter: 'data.uuid' },
            {headerName: 'ID', valueGetter: 'data.identifier' },
            {headerName: 'First Name', valueGetter: 'data.firstName' },
            {headerName: 'Last Name', valueGetter: 'data.lastName' },
            {headerName: 'Gender', field: 'gender' },
            {headerName: 'Age', field: 'age' }
        ];
    }


    parseResults(results) {
        console.log("parseResults callback function, length =  " + results.length);
        let patients = [];
        for (const item of results) {
            let patient = {
                uuid: item.uuid,
                id: item.id,
                firstName: item.person.names[0].givenName,
                lastName: item.person.names[0].familyName,
                gender: item.person.gender,
                age: item.person.age,
                identifier: item.identifiers[0].identifier,
                checkedInTime: null,
                birthdate: item.person.birthdate
            };
            patients.push(patient);
        }
        return patients;
    };

    render() {
        return(
            <PatientSearch columnDefs={ this.columnDefs } parseResults={ this.parseResults.bind(this) }/>
        );
    }
};

export default SearchPatient;
