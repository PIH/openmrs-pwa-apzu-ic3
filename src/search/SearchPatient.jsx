import React from 'react';
import { PatientSearch } from '@openmrs/react-components';
import { push } from 'connected-react-router';
import patientActions from '../patient/patientActions';
import IdentifierFilters from '../screening/IdentifierFilters';
import utils from '../utils'


let SearchPatient = (props) => {
  return (
    <div>
      <PatientSearch
        AdditionalFilters={IdentifierFilters}
        rowSelectedActionCreators={[
          patientActions.getIC3PatientScreeningData,
          () => push('/checkin/checkInPage')
        ]}
        getPatientIdentifiers={utils.getPatientIdentifiers}
        title="Search for Patient"
      />
    </div>
  );
};

export default SearchPatient;
