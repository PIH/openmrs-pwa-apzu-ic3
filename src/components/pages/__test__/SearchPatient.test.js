import React from 'react'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import SearchPatient from '../SearchPatient';

let props, mountedComponent, store;

const mockStore = configureMockStore();
const searchPatient = () => {
    if (!mountedComponent) {
        mountedComponent = shallow(
            <Provider store={store}>
                <SearchPatient {...props} />
            </Provider>);
    }
    return mountedComponent;
};

describe('Component: searchPatient', () => {
    beforeEach(() => {
        props = {};
        store = mockStore(
            {
                selected: {
                    patient: {
                        id: 1,
                        lastName: "Wilkin",
                        firstName: "Dessin",
                        gender: "M",
                        age: 25,
                    }
                }
            });
        mountedComponent = undefined;
    });

    it('renders properly', () => {
        expect(toJson(searchPatient())).toMatchSnapshot();
        //expect(searchPatient().prop('className')).to.equal(1);
    });

});

