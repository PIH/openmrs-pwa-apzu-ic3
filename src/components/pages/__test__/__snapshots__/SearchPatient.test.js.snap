// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: searchPatient renders properly 1`] = `<Connect(SearchPatient) />`;
